customerio/customerio-ios
# fix: universal links deep links open host app

Bug fix in regards to [this comment](https://github.com/customerio/customerio-ios/pull/265/files#r1104311274) about universal links deep links not working in production. 

To prevent this human error from happening again, I refactored the code to no longer use a locally scoped variable. 

- [ ] [Deploy docs update](https://github.com/customerio/docs/pull/938) once code gets deployed. 

# QA
RH branch `levi/universal-links-fix` created for code reviewer on team to QA this PR before we merge it. Current build is `push.529014775`

Steps to QA
* Customer.io broadcast "test push deep links" has been created to allow easier testing. Modify the broadcast to a Segment that targets just your devices. See "Levi segment" as a good example to go off of. 
* Make sure to login to RH app and wait for push device token to register. Your device should now exist in the Segment the testing broadcast is using. 
* Trigger the broadcast. Your ios device should now get 2 push notifications. 
* Select each push notification one at a time. Both push notifications are expected to launch the RH app -> open switch workspace screen -> fill in the text fields on the screen to either "app", "scheme" or "universal", "link" depending on what push you opened. **The system browser should not open when you open the push notification at anytime. Only the host app should open**.
